wgpu/src/lib.rs

```diff
@@ -2363,6 +2387,22 @@ impl Display for RequestDeviceError {
 
 impl error::Error for RequestDeviceError {}
 
+/// [`Instance::create_surface()`] or a related function failed.
+#[derive(Clone, PartialEq, Eq, Debug)]
+#[non_exhaustive]
+pub struct CreateSurfaceError {
```

Comment:
Let's make this an enum

Reply:
If we make it an enum, it needs at least one variant, which will have public fields if it has any fields. What should that variant be called, and what fields should it have, given that currently no specific information is available?

Something like this as an explicitly “some platform-specific situation” variant?

```rust
#[non_exhaustive]
pub struct CreateSurfaceError {
    Platform(String),
}
```

I was hoping that the totally-private struct would be adequate as a placeholder until we could fix the lack of information.

Reply:
Reasonable enough.